website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,159 @@
+---
+layout: docs
+page_title: Upgrading Specific Versions
+description: >-
+  Consul api gateway versions v0.2.x introduced a change to namespace backend ref
+  enforcement that will potentially break any routes created with a cross namespace in
+  v0.1.0. This documents the flow for finding and remediating any potentially
+  broken routes
+---
+
+# Upgrading Specfic Versions
+
+
+## Consul API Gateway v0.1.0
+
+This guide explains how to best upgrade a consul-api-gateway deployment that is currently
+running v0.1.0 to all future versions. v0.2.0 introduced a breaking change that
+causes routes defined with a BackendRef defined in a different namespace to
+require a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy)
+that explicitly allows traffic from the routes namespace to the backendrefs namespace.
+
+
+
+## Requirements
+
+- The consul-api-gateway should be running version v0.1.0, if the version is different, follow the normal upgrade path
+- [jq](https://stedolan.github.io/jq/download/) is installed on the users CLI
+
+Note, as of writing, this is the only upgrade path documented as v0.1.0 is the only currently released version. Normal upgrade path will be documented
+in future releases
+
+## Assumptions
+
+This guides makes the following assumptions:
+
+- You have the ability to run Kubectl cli commands
+- Your kubectl config is already configured to point at the cluster with the installation you are upgrading
+- You have HTTPRoute.read, TCPRoute.read and ReferencePolicy.create rights on your kubernetes cluster
+
+
+## Procedures
+
+**1.** Optional, double check the current version of the consul-api-gateway with the following command
+
+```
+kubectl get deployment --namespace consul consul-api-gateway-controller -o json | jq '.spec.template.spec.containers[] | select(.name="api-gateway-controller") .image'
+```
+
+If the output looks as follows
+
+```
+"hashicorp/consul-api-gateway:0.1.0"
+```
+
+ continue following this upgrade path.
+
+**2.** Run the following command to retrieve all routes defined with a cross namespace:
+
+```
+kubectl get HttpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  ' && kubectl get TcpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  '
+
```

Comment:
I'm not entirely sure without testing this myself, but I _think_ you can simplify this command by performing a simultaneous get of both `TcpRoute` and `HttpRoute`, then parsing the items from a single array.

```suggestion
kubectl get HttpRoute,TcpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )} '
```

Reply:
That worked, good call, thank you!